src/lib/sidenav/drawer.ts

```diff
@@ -246,13 +252,15 @@ export class MatDrawer implements AfterContentInit, AfterContentChecked, OnDestr
 
   /** Traps focus inside the drawer. */
   private _trapFocus() {
-    this._focusTrap.focusInitialElementWhenReady().then(hasMovedFocus => {
-      // If there were no focusable elements, focus the sidenav itself so the keyboard navigation
-      // still works. We need to check that `focus` is a function due to Universal.
-      if (!hasMovedFocus && typeof this._elementRef.nativeElement.focus === 'function') {
-        this._elementRef.nativeElement.focus();
-      }
-    });
+    if (this.autoFocus) {
```

Comment:
does `_restoreFocus` need new logic as well?

Reply:
Hmm, good point. I had forgotten that it has some extra functionality on top of just restoring focus. It's changed.